packages/shared/src/util/features.ts

```diff
@@ -114,3 +116,54 @@ export function validateFeatureValue(
 
   return value;
 }
+
+type StaleFeatureReason =
+  | "error"
+  | "draft-state"
+  | "no-active-envs"
+  | "no-rules"
+  | "no-active-exps"
+  | "all-exps-onesided";
+
+export function isFeatureStale(
+  feature: FeatureInterface,
+  linkedExperiments: ExperimentInterface[] | undefined = []
+): { stale: boolean; reason?: StaleFeatureReason } {
+  if (feature.linkedExperiments?.length && !linkedExperiments.length) {
+    // eslint-disable-next-line no-console
+    console.error("isFeatureStale: linkedExperiments not provided");
+    return { stale: false, reason: "error" };
+  }
+
+  const twoWeeksAgo = subWeeks(new Date(), 2);
+  // TODO why is dateUpdated a string?
+  const stale = new Date(feature.dateUpdated) < twoWeeksAgo;
```

Comment:
There is a `getValidDate` function that does a little more validation and sanity checking